### ISSUE TITLE
max-line-length optionally disabled for import statements

### ISSUE THREAD

**Issue by michalstocki** [palantir/tslint#1123]
We use 120-chars long `max-line-length` rule, but while the application grows, we often encounter cases where the 120 chars is to small for import statements.
For some time, we've used a workaround, breaking line after `from` keyword:

``` typescript
import {ReallyLongClassNameLikeInLargeApplications}
  from '../../../../nested/in/the/abyss/of/our/large/directory/structure/deep/deep/deep/ReallyLongClassNameLikeInLargeApplications.ts'
```

And this worked till one of the latest updates of WebStorm IDE, when we got awesome "optimize imports" flag in "Code > Reformat Code" dialog, which removes unused imports, duplication etc., but also removes our line-break in the `import … from` statement.
This became dangerous when I tried to configure IDE to reformat my code before each commit automatically. Implicit code reformat, causes breaking TSLint rule just before the commit – which brings the broken code, to the repo.

**Comment by basst314** [comment]
any updates on this? @palantirtech

**Comment by michalstocki** [comment]
It's worth to mention that ESLint has an `ignorePattern` option that allows to address the issue of long imports: http://eslint.org/docs/rules/max-len#ignorepattern

**Comment by everedifice** [comment]
Any updates? ignoring all line by commenting is nonsense.

I think adding `ignorePattern` option is right direction.

**Comment by lubos-bistak** [comment]
Is there any update for this issue?
I am using tslint and need some way how to ignore long import from being violated through `max-line-length`.
Is there at least some workaround?

**Comment by ajafff** [comment]
An option to specify an ignore pattern just landed in #3099
